lamindb/curators/core.py

```diff
@@ -358,24 +358,56 @@ def save_artifact(
         )
 
 
+def _is_type_or_list_of_type(value, expected_type):
```

Comment:
Why prefix this with an underscore but not `check_dtype`? I'd say this shouldn't be prefixed for consistency.